# OMNI (Orthologous group Merger for Numerous Inference tools)


## Repository Structure

```
.
â”œâ”€â”€ Fasta/                           # Raw FASTA files used in the analysis
â”œâ”€â”€ Output/                          # Output files from orthogroup and phylogeny analyses
â”œâ”€â”€ fasta_protein/                   # Protein sequences in FASTA format
â”œâ”€â”€ particule_species_Busco.csv      # BUSCO results for different species
â”œâ”€â”€ phylogenie_apicomplexa.nwk       # Phylogenetic tree in Newick format
â”œâ”€â”€ prot_test.csv                    # Protein test data
```

## Requirements

- Python 3.8+<p align="center"><img alt="Made with Python" src="https://img.shields.io/badge/Made%20with-Python-1f425f.svg?color=%23539fc9"></p>
- EMBOSS (transeq dans le `$PATH`)
- [BUSCO](https://busco.ezlab.org/)
- [SonicParanoid](https://github.com/fenderglass/SonicParanoid)
- [OrthoFinder](https://github.com/davidemms/OrthoFinder)
- Orthologer
- `pandas`, `argparse`, `os`, `glob`, `shutil`, etc.

---

## ðŸ‘¤ Author

**Romain Daguerre**
UMR 7245 - CNRS MNHM - Parasites et Protistes Libres
Master Bioinformatique â€” UniversitÃ© Paris CitÃ©


---
